docs/source/projects.rst

```diff
@@ -349,103 +349,170 @@ useful if you quickly want to test a project in your existing shell environment.
+     file.
+   - The Docker repository referenced by ``repository-uri`` in your backend configuration file.
+   - The `Kubernetes context
+     <https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#context>`_
+     referenced by ``kube-context`` in your backend configuration file.
+
+   MLflow expects these resources to be accessible via the
+   `docker <https://docs.docker.com/engine/reference/commandline/cli/>`_ and
+   `kubectl <https://kubernetes.io/docs/reference/kubectl/kubectl/>`_ CLIs before running the
+   Project.
+
+4. Finally, run the Project using the MLflow Projects CLI or
+   :py:func:`Python API <mlflow.projects.run>`, specifying your Project URI and the path to your
+   backend configuration file. For example:
+
+   .. code-block:: bash
+
+    mlflow run <project_uri> --backend kubernetes --backend-config examples/docker/kubernetes_config.json
+
+   where ``<project_uri>`` is a Git repository URI or a folder.
+
+.. _kubernetes_execution_job_templates:
+
+Job Templates
+~~~~~~~~~~~~~
+
+MLflow executes Projects on Kubernetes by creating `Kubernetes Job resources
+<https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/>`_.
+MLflow creates a Kubernetes Job for an MLflow Project by reading a user-specified
+`Job Spec
+<https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#writing-a-job-spec>`_.
+When MLflow reads a Job Spec, it formats the following fields:
+
+- ``metadata.name`` Replaced with a string containing the name of the MLflow Project and the time
+  of Project execution
+- ``spec.template.spec.container[0].name`` Replaced with the name of the MLflow Project
+- ``spec.template.spec.container[0].image`` Replaced with the URI of the Docker image created during
+  Project execution. This URI includes the Docker image's digest hash.
+- ``spec.template.spec.container[0].command`` Replaced with the Project entry point command
+  specified when executing the MLflow Project.
+
+The following example shows a simple Kubernetes Job Spec that is compatible with MLflow Project
+execution. Replaced fields are indicated using bracketed text.
+
+.. rubric:: Example Kubernetes Job Spec
 
-- ``metadata.name``;
-
-- ``spec.template.spec.container[0].name``;
-
-- ``spec.template.spec.container[0].image``;
-
-- ``spec.template.spec.container[0].command``;
-
-
-.. code-block:: json
+.. code-block:: yaml
 
   apiVersion: batch/v1
   kind: Job
   metadata:
-    name: ""
+    name: "{replaced with MLflow Project name}"
     namespace: mlflow
   spec:
     ttlSecondsAfterFinished: 100
     backoffLimit: 0
     template:
       spec:
         containers:
-        - name: "name"
-          image: "image"
-          command: ["cmd"]
+        - name: "{replaced with MLflow Project name}"
+          image: "{replaced with URI of Docker image created during Project execution}"
+          command: [{replaced with MLflow Project entry point command}]
```

Comment:
Does this example work out of the box? I think this needs to be a string right?

Reply:
Actually can you make https://github.com/mlflow/mlflow/blob/master/examples/docker/kubernetes_job_template.yaml match this then?

Reply:
It does, but I've changed it to a string with *braces* based on our discussion. I also updated the template in `examples/docker`